handlers/validator.go

```diff
@@ -308,6 +185,20 @@ func Validator(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Check if the validator has balance data in the finalized epochs
+	// balances, err := db.BigtableClient.GetValidatorBalanceHistory([]uint64{index}, latestFinalized, latestFinalized)
+	var balances map[uint64][]*types.ValidatorBalance
+	if err != nil {
+		http.Error(w, "error getting validator balance data", 404)
+		return
+	}
+	if _, ok := balances[index]; !ok {
+		// An index would be available but there is no balance data in the finalized epochs
+		// so treat it as if the index is not available
+		handleValidatorWithNoIndex(w, r, validatorPageData.PublicKey, index, &validatorPageData, data)
```

Comment:
This code triggers for every single validator. You initialize a map (which is empty) and then, if the map doesn't have a specific entry (which it always won't), you render a template without the index? I don't get it.

Reply:
Sorry about that. Apparently some test scenario wasn't reverted properly. Fixed.